django/db/models/fields/related.py

```diff
@@ -1885,6 +1885,13 @@ def __init__(self, to, to_field=None, related_name=None, related_query_name=None
             # be correct until contribute_to_class is called. Refs #12190.
             to_field = to_field or (to._meta.pk and to._meta.pk.name)
 
+        if on_delete is None:
+            warnings.warn(
+                "Implicit default of ForeignKey.on_delete has been deprecated. "
```

Comment:
I've looked and haven't found nothing in `ForeignObject` that makes it default to `CASCADE`, so I've decided to leave it alone. But I can add the very same check there if you rather.